x/accounts/genesis.go

```diff
@@ -80,7 +80,14 @@ func (k Keeper) ImportState(ctx context.Context, genState *v1.GenesisState) erro
 }
 
 func (k Keeper) importAccount(ctx context.Context, acc *v1.GenesisAccount) error {
-	// TODO: maybe check if impl exists?
+	// Check if the account type exists in the registered accounts
+	_, ok := k.accounts[acc.AccountType]
+	if !ok {
+		// If the account type does not exist, log a warning and continue without importing the account
+		fmt.Printf("Warning: Account type %s not found in the registered accounts. Skipping import.\n", acc.AccountType)
```

Comment:
I think we should return an error. You don't want to just almost silently skip importing accounts

Reply:
```
+++ b/x/accounts/genesis.go
@@ -83,9 +83,8 @@ func (k Keeper) importAccount(ctx context.Context, acc *v1.GenesisAccount) error
        // Check if the account type exists in the registered accounts
        _, ok := k.accounts[acc.AccountType]
        if !ok {
-               // If the account type does not exist, log a warning and continue without importing the account
-               fmt.Printf("Warning: Account type %s not found in the registered accounts. Skipping import.\n", acc.AccountType)
-               return nil
+               // If the account type does not exist, return an error
+               return fmt.Errorf("account type %s not found in the registered accounts", acc.AccountType)
        }
```
If I understand correctly, you want to modify the code to return an error if the GenesisAccount type is not found in the registered accounts. Is that correct?